x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -130,17 +148,53 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
             id: alertId,
           };
 
+          /**
+           * Data Views Logic
+           * Use of data views is supported for all rules other than ML.
+           * Rules can define both a data view and index pattern, but on execution:
+           *  - Data view is used if it is defined
+           *    - Rule exits early if data view defined is not found (ie: it's been deleted)
+           *  - If no data view defined, falls to using existing index logic
+           */
+          if (!isMachineLearningParams(params)) {
+            try {
+              const { index, runtimeMappings: dataViewRuntimeMappings } = await getInputIndex({
+                index: params.index,
+                services,
+                version,
+                logger,
+                ruleId: params.ruleId,
+                dataViewId:
+                  isEqlParams(params) ||
+                  isQueryParams(params) ||
+                  isThresholdParams(params) ||
+                  isThreatParams(params)
+                    ? params.dataViewId
+                    : undefined,
```

Comment:
Is the intent here to support dataViewId for saved query rules or not? I see the dataViewId param on the HTTP API for saved query rules, but not on the internal rule schema - which is why the extra 
```
                  isEqlParams(params) ||
                  isQueryParams(params) ||
                  isThresholdParams(params) ||
                  isThreatParams(params)
```
is needed here. If we add `dataViewId` to saved query params, then the `if (!isMachineLearningParams(params))` check above will narrow the type so we can access `params.dataViewId` on any remaining type of `params` in here.

Reply:
resolved here: b535d5c117c2d0530d760e76a2d0cf4ad3f0ae36